src/org/zaproxy/zap/extension/exportreport/ExportReportAPI.java

```diff
@@ -0,0 +1,184 @@
+import org.parosproxy.paros.Constant;
+import org.zaproxy.zap.extension.api.ApiAction;
+import org.zaproxy.zap.extension.api.ApiException;
+import org.zaproxy.zap.extension.api.ApiException.Type;
+import org.zaproxy.zap.extension.api.ApiImplementor;
+import org.zaproxy.zap.extension.api.ApiResponse;
+import org.zaproxy.zap.extension.api.ApiResponseElement;
+import org.zaproxy.zap.extension.api.ApiResponseList;
+import org.zaproxy.zap.extension.api.ApiView;
+import org.zaproxy.zap.extension.exportreport.filechooser.FileList;
+import org.zaproxy.zap.extension.exportreport.filechooser.FileType;
+
+import net.sf.json.JSONObject;
+
+/**
+ * The API for export report.
+ *
+ * AUTHOR: GORAN SARENKAPA - JordanGS
+ * SPONSOR: RYERSON UNIVERSITY
+ *
+ */
+public class ExportReportAPI extends ApiImplementor {
+
+    private static final Logger logger = Logger.getLogger(ExportReportAPI.class);
+
+    private static final String PREFIX = "exportreport";
+
+    private static final String ACTION_GENERATE = "generate";
+    private static final String VIEW_FORMATS = "formats";
+
+    private static final String ACTION_PARAM_ABSOLUTE_PATH = "absolutePath";
+    private static final String ACTION_PARAM_FILE_EXTENSION = "fileExtension";
+    private static final String ACTION_PARAM_SOURCE_DETAILS = "sourceDetails";
+    private static final String ACTION_PARAM_ALERT_SEVERITY = "alertSeverity";
+    private static final String ACTION_PARAM_ALERT_DETAILS = "alertDetails";
+
+    private ExtensionExportReport extension;
+
+    public ExportReportAPI(ExtensionExportReport extension) {
+        super();
+        this.extension = extension;
+
+        this.addApiAction(
+                new ApiAction( ACTION_GENERATE,
+                new String[] { ACTION_PARAM_ABSOLUTE_PATH,
+                               ACTION_PARAM_FILE_EXTENSION,
+                               ACTION_PARAM_SOURCE_DETAILS,
+                               ACTION_PARAM_ALERT_SEVERITY,
+                               ACTION_PARAM_ALERT_DETAILS }));
+        this.addApiView(new ApiView(VIEW_FORMATS));
+    }
+
+    @Override
+    public String getPrefix() {
+        return PREFIX;
+    }
+
+    @Override
+    public ApiResponse handleApiAction(String name, JSONObject params) throws ApiException {
+        if (logger.isDebugEnabled()) logger.debug("Request for handleApiAction: " + name + " (params: " + params.toString() + ")");
+
+        switch (name) {
+        case ACTION_GENERATE:
+            String absolutePath = params.getString(ACTION_PARAM_ABSOLUTE_PATH);
+
+            String fileExtension = params.getString(ACTION_PARAM_FILE_EXTENSION);
+
+            if (!extension.canWrite(absolutePath)) {
+                logger.error(Constant.messages.getString("exportreport.message.console.error.file.writable", absolutePath));
+                return ApiResponseElement.FAIL;
+            }
+
+            boolean valid = false;
+
+            if (fileExtension.length() > 0) {
```

Comment:
This check is not really needed, since the parameter is mandatory core code already takes care to validate that's defined and non-empty.
Although, shouldn't be validated that's one of the supported formats?